pyhocon/config_parser.py

```diff
@@ -207,6 +245,17 @@ def convert_number(tokens):
             except ValueError:
                 return float(n)
 
+        def convert_period(tokens):
+
+            period_value = int(tokens.value)
+            period_identifier = tokens.unit
+
+            period_unit = [single_unit for single_unit, values
+                           in get_supported_period_type_map().items()
+                           if period_identifier in values][0]
```

Comment:
nit: maybe use `next()`:
```python
period_unit = next(
  (
     single_unit
     for single_unit, values in get_supported_period_type_map().items()
     if period_identifier in values
  ),
  None
)

if period_unit:
    return period(period_value, period_unit)
else:
    raise ConfigException('Period {identifier} is not supported: {loc}: {instring}', identifier= period_identifier, loc=loc, instring=instring)

```

and change the function signature to:

```python
def convert_period(instring, loc, token)
```

Maybe precompute or cache `get_supported_period_type_map` in a class variable like `supported_period_type_map` so we do not need to recompute it every time.

Reply:
fixed the next() issues. cached the supported period type map

as for the optimization, we could extend that to the bulk of the parse method instead of just the period parsing. but that should be a separate task.

Reply:
Thank you @gabis-precog! It looks great! I'm going to merge it tonight unless you want to update it a little more.